### PR TITLE
Canvas Class Migration

### PR DESCRIPTION
**Pull request by SkamDart**
Don't merge yet just moved functions from output canvas to canvas.js

### CODE REVIEW

File: public/js/project/canvas/canvas.js [hunk 1]
```diff
@@ -79,46 +79,123 @@ function canvas(settings) {
...
+  }
+  
   /**
   *
   *
   */
-
   function proximity(outputVector, target) {
 
-  }
```
Comment (cdart2) [inline public/js/project/canvas/canvas.js]: Move below inside this function

File: public/js/project/canvas/canvas.js [hunk 2]
```diff
@@ -79,46 +79,123 @@ function canvas(settings) {
...
   }
 
   function drawVector(vector) {
-
-  }
-
-  function drawTarget(target) {
-
-  }
+    vectorItem = new Group([
```
Comment (SkamDart) [inline public/js/project/canvas/canvas.js:89]: Not usable code but need to differentiate between update and draw.

File: public/js/project/canvas/canvas.js [hunk 3]
```diff
@@ -79,46 +79,123 @@ function canvas(settings) {
...
       aLine.strokeColor = 'black';
 
-      if (i == num_rectangles_tall / 2) {
+      if (i === num_rectangles_tall / 2) {
         aLine.strokeWidth = 5;
       }
     }
   }
 
   function drawVector(vector) {
```
Comment (SkamDart) [inline public/js/project/canvas/canvas.js:88]: PaperScript inside here?

File: public/js/project/canvas/canvas.js [hunk 4]
```diff
@@ -79,46 +79,123 @@ function canvas(settings) {
...
+    }
+  }
 
   /**
-  *
-  *
+  * @PARAM target object as described in target.js
+  * @RETURNS nothing
   */
+  function drawTarget(target) {
```
Comment (SkamDart) [inline public/js/project/canvas/canvas.js:190]: where is this being drawn on?

File: public/js/project/canvas/canvas.js [hunk 5]
```diff
@@ -79,46 +79,123 @@ function canvas(settings) {
...
+    targetPath = new Path.Circle(new Point(target.x, target.y), 10);
+    targetPath.fillColor = '#e5e5ff';
+  }
 
   function drawTargets(targets) {
```
Comment (SkamDart) [inline public/js/project/canvas/canvas.js:195]: might make sense to add canvas object

File: public/js/project/canvas/canvas.js [hunk 6]
```diff
@@ -79,46 +79,123 @@ function canvas(settings) {
...
   /**
   *
   *
   */
-
   function proximity(outputVector, target) {
 
-  }
 
+    if (isClose(targetX, targetY, 500)) {
```
Comment (SkamDart) [inline public/js/project/canvas/canvas.js:135]: update function signature to match what is above.

File: public/js/project/canvas/canvas.js (hunk 4 above)
Comment (cmalk) [inline public/js/project/canvas/canvas.js:190]: The canvas associated to the paperscript scope associated to that instance of "canvas." A mess, I know...

File: public/js/project/canvas/canvas.js (hunk 5 above)
Comment (josephmilla) [inline public/js/project/canvas/canvas.js:195]: There might be a better way to change the color of the targets based on proximity. Bitwise shifting?

**Pull request closed without merging**